.github/actions/setup-binary-builds/action.yml

```diff
@@ -48,15 +53,26 @@ runs:
         shell: bash
         env:
           REPOSITORY: ${{ inputs.repository }}
+          IS_MANYLINUX2_28: ${{ inputs.is-manylinux2-28 }}
         run: |
           set -euxo pipefail
           rm -rf "${REPOSITORY}"
+
       - uses: actions/checkout@v4
+        if: ${{ env.IS_MANYLINUX2_28 == 'true' }}
         with:
           repository: ${{ inputs.repository }}
           ref: ${{ inputs.ref }}
           submodules: ${{ inputs.submodules }}
           path: ${{ inputs.repository }}
+
+      - uses: atalman/checkout-action@main
```

Comment:
Lets keep both ``actions/checkout@v4`` and ``atalman/checkout-action@main``. I believe this action is  run on Windows and MacOS. I see no reason to debug this on these OS's